gflownet/envs/amp.py

```diff
@@ -202,34 +211,38 @@ def true_density(self, max_states=1e6):
     def state2oracle(self, state: List = None):
         return "".join(self.state2readable(state))
 
-    def statebatch2oracle(self, state_batch: List[List]):
-        """
-        Prepares a sequence in "GFlowNet format" for the oracles.
-
-        Args
-        ----
-        state_list : list of lists
-            List of sequences.
-        """
-        state_oracle = [self.state2oracle(state) for state in state_batch]
-        return state_oracle
-
     def get_max_traj_len(
         self,
     ):
         return self.max_seq_length / self.min_word_len + 1
 
+    def statebatch2oracle(
+        self, states: List[TensorType["max_seq_length"]], bos_idx=None
```

Comment:
Is the input `states` a list of tensors as the typing indicates? Is this ever used?

Reply:
Yeah, statebatch is a list of tensors for AMP (as in other envs it is a list of lists). Yes it is used to get the true energies after the AL round as the trained gflownet samples a list of states (tensors)